robotd/devices.py

```diff
@@ -128,28 +130,43 @@ def status(self):
         return {'temperature': temp_milli_degrees / 1000}
 
 
+class NoZoneFound(ValueError):
+    """Indicates that a search for a zone file failed."""
+    pass
+
+
 class GameState(Board):
     """
     State storage for the game, keeps a store of everything it has received.
     """
 
+    FILE_GLOB = '/media/usb?/zone-?'
+    FILE_REGEX = re.compile('zone-(\d)')
+
     # define the name od the board
     board_type_id = 'game'
     create_on_startup = True
 
-    def __init__(self):
-        super().__init__({})
-        self.state = {'zone': 0, 'mode': 'development'}
-
     @classmethod
     def name(cls, node):
         return 'state'
 
-    def command(self, cmd):
-        self.state.update(cmd)
+    def find_zone(self):
+        for candidate_path in glob.glob(self.FILE_GLOB):
+            match = self.FILE_REGEX.search(candidate_path)
```

Comment:
It was also discussed that we should exclude any matches which have a `main.py` on, incase teams we doing weird things. I think we should still implement this?

Reply:
We did, I forgot.

This actually introduces knowledge of `runusb` into `robotd` which isn't really ideal, though probably ok for now.

Reply:
61688d89007bb76c1649242e18a1ed5069091a24